plugins/contactpressuresensorsarray/src/ContactPressureSensorsArrayDriver.cpp

```diff
@@ -0,0 +1,516 @@
+        loadCellLoc.push_back(loadCellZ->get(i).asDouble());
+        this->m_loadCellLocations.push_back(loadCellLoc);
+     }
+  
+     m_contactNormalForces.resize(m_loadCellLocations.size());  
+     if (!this->prepareMappingMatrix())
+     {
+         return false;
+     }
+    
+    return true;
+}
+
+bool GazeboYarpContactPressureSensorsArrayDriver::prepareMappingMatrix()
+{
+    // get the size of the load cell locations
+    int n_cells = this->m_loadCellLocations.size();
+  
+    // Consider w = Af ; where w is the wrench of size 3x1 containing normal force, tangential and radial torques
+    // A is the matrix that maps the array of pressure sensor normal forces f to an resultant wrench w acting 
+    // at the contact. However, we ae solving the inverse problem of mapping a wrench acting at the contact link
+    // to the presure sensor array normal forces, hence we need to obtain the pseudo inverse through
+    // f = (A+)w ; where (A+) is the pseudo inverse. Mapping an equivalent wrench to a number of pressure sensor 
+    // normal forces has infinite solutions, but considering the pseudo inverse gives us the solution with minimized norm.
+    // Hence it is not necessary that the array of pressure sensors simulate the same forces at corresponding locations
+    // similar to the real robot. However, the resultant wrench from the pressure sensor array of normal forces should be valid.
+    // The matrix A is the function of loadCell locations (x, y) with respect to the link coordinate frame
+    yarp::sig::Matrix A; 
+    A.resize(N_AXIS ,n_cells);
+  
+    // Add the respective columns to the mapping matrix
+    for (int i = 0; i < n_cells; i++)
+    {
+        yarp::sig::Vector columnMap;
+        columnMap.clear();
+        columnMap.push_back(1);
+        columnMap.push_back(this->m_loadCellLocations[i][1]); // r_y : y-intercept of the contact point with respect to link origin
+        columnMap.push_back(-this->m_loadCellLocations[i][0]); // -r_x : x-intercept of the contact point with respect to link origin
+        
+        if (!A.setCol(i, columnMap))
+        {
+            yError() << "GazeboYarpContactPressureSensorsArrayDriver Failed: Failed to form mapping matrix";
+            return false;
+        }
+    }
+  
+    if (A.rows() != N_AXIS || A.cols() != m_loadCellLocations.size())
+    {
+        yError() << "GazeboYarpContactPressureSensorsArrayDriver Failed: Mismatched dimensions in the mapping matrix";
+        return false;
+    }
+ 
+    // Compute the pseudo-inverse to determine the mapping matrix
+    this->m_mapWrenchtoNormalForce = yarp::math::pinv(A);
+
+    return true;
+}
+
+// gazebo::math::Quaternion method GetAsMatrix() is deprecated from version 8.0
+// and ignition::math::Quaternion does not support this method, so porting it
+// from gazebo::math just to support double type values, change if necessary
+ignition::math::Matrix3d GetQuaternionAsMatrix3d(ignition::math::Quaterniond q) 
```

Comment:
I guess you can use the https://osrf-distributions.s3.amazonaws.com/ign-math/api/1.0.0/classignition_1_1math_1_1Matrix3.html#a03d18617b019eb137c318bd60d653794 Quaternion-based Matrix3d constructor for this.

Reply:
Thanks, great!